lib/mayaUsd/utils/utilFileSystem.cpp

```diff
@@ -313,6 +313,54 @@ void UsdMayaUtilFileSystem::unmarkPathAsPostponedRelative(
     }
 }
 
+template <typename TypePolicy>
+void updatePathList(
+    SdfListProxy<TypePolicy>                list,
+    const PostponedRelativePaths::iterator& layerEntry,
+    const std::string&                      anchorDirStr)
+{
+    for (auto proxy : list) {
+        typename TypePolicy::value_type item = proxy;
+        ghc::filesystem::path           filePath(item.GetAssetPath());
+        filePath = filePath.lexically_normal();
+
+        auto it = layerEntry->second.find(filePath);
+        if (it == layerEntry->second.end()) {
+            continue;
+        }
+
+        item.SetAssetPath(UsdMayaUtilFileSystem::getPathRelativeToDirectory(
+            filePath.generic_string(), anchorDirStr));
+        proxy = item;
+    }
+}
+
+void updatePostponedRelativePathsForPrim(
+    const SdfPrimSpecHandle&                primSpec,
+    const PostponedRelativePaths::iterator& layerEntry,
+    const std::string&                      anchorDirStr)
+{
+    for (const SdfPrimSpecHandle& child : primSpec->GetNameChildren()) {
```

Comment:
Question: does this only return child prim spec that are in the same layer or can it return prim spec from other layers? What about references? Can it traverse references?

IDK, for example, what happens if the user adds a payload to a prim that only lives in a reference? Will it be found?

Reply:
Mmm, we're only trying to update things that are inside the layer that just got saved. So I guess my question is irrelevant.